### ISSUE TITLE
solve should use syzygies over rings where it is not implemented

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#1051]
I believe this is a bug:
```
restart
R = ZZ/32003[x,y,z]
A = matrix{{x,y,z,0,0,0,z,y,x^2,0,0,0},{0,x,0,y,z,0,0,z,0,y,x^2,0},{0,0,x,0,y,z,0,0,z,0,y,x^2}}
v = transpose matrix{{y^2-x*z,0,0}}

v - A * solve(A,v) == 0 -- false

s = syz (A | v);
v + A * (s_{9})^{0..11} == 0 -- true, as expected
```

**Comment by DanGrayson** [comment]
The documentation for "solve" says the matrix should be over a field.  Maybe the bug is for it not to give an error message.

**Comment by mahrud** [comment]
Ah, I didn't notice that. Still, when the systems is solvable using syzygies, generalizing `solve` should be easy. I can make a PR to do that if you'd like.

**Comment by DanGrayson** [comment]
@mikestillman - what do you think?  It's your code.

**Comment by pepperhdj** [comment]
I think this was addressed in [this](https://github.com/Macaulay2/M2/pull/1141) pull request.

**Comment by dylanpeifer** [comment]
Yes, the pull request changes `solve` to give an error. Generalizing to use syzygies might still be useful.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

The defect reported here is fixed, so this issue is being retitled to the question that is
actually still open.

**`solve` now raises an error instead of returning a wrong answer.** The same input gives

```
error: solve not implemented over ring R
```

from `m2/mutablemat.m2:216`. That is what @pepperhdj and @dylanpeifer identified in 2020 —
[#1141](https://github.com/Macaulay2/M2/pull/1141) — and it is the change @DanGrayson asked for
above: *"maybe the bug is for it not to give an error message."*

**What is left is the generalisation, and it is an offer nobody answered.** @mahrud wrote in 2019:

> Still, when the system is solvable using syzygies, generalizing `solve` should be easy. I can
> make a PR to do that if you'd like.

@DanGrayson put that to @mikestillman, and the thread stops there. @dylanpeifer restated it in 2020
— *"Generalizing to use syzygies might still be useful"* — and that also went unanswered. The
working method is in the issue body above: `syz (A | v)` finds the solution that `solve` will not.

So the title has been changed from *"Bug in solve"* to *"solve should use syzygies over rings where
it is not implemented"* — the wrong answer is gone, and what this tracks now is a capability that
was offered and never taken up. The new title quotes the error text so that searching for it lands
here. Nothing else about the issue has been altered.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
